File: base/python/sort.py
import os

def compute():
    # size of the array to sort
    size = int(os.environ['ARRAY_SIZE'])

    # create an input array with unsorted values
    arr = [0] * size
    incr = 1
    for i in range(size):
        arr[i] = -arr[i-1] + incr
        incr = -incr

    # sort the array
    arr.sort()

File: base/go/sort.go
package main

import (
	"os"
	"sort"
	"strconv"
)

func compute() {

	// size of the array to sort
	size, _ := strconv.Atoi(os.Getenv("ARRAY_SIZE"))

	//create an input array with unsorted values
	input := make([]int, size)
	incr := 1
	for i := 1; i < size; i++ {
		input[i] = -input[i-1] + incr
		incr = -incr
	}

	// sort the array
	sort.Ints(input)
}
